mozart++/string/string.hpp

```diff
@@ -33,7 +33,7 @@ namespace mpp {
     public:
         static constexpr size_t npos = ~size_t(0);
 
-        using iterator = const char *;
+        using iterator = char *;
```

Comment:
This `iterator` should be `const char *`, as the `string_ref` is designed to be an **immutable** string reference and only holds the reference, which may not be modified.

Codes like this are not allowed:
```cpp
const char *text = "hello world";
string_ref ref = text;
for (s : ref.bytes()) {
    *s = *s + 1;
}
```
Here, the `ref` holds a reference to read-only text, and this code will definitely cause a `segmentation fault`.